docs/ops/doc/ParameterConversion.rst

```diff
@@ -0,0 +1,141 @@
+========================================
+Parameter Conversion for Developers
+========================================
+
+In this example, we show how Ops developers can implement parameter conversion to enable seamless interoperability between Ops utilizing different data structures.between
+
+Basics
+======
+
+A key :ref:`value <driving-values>` of SciJava Ops is flexibility, and flexibility is (in part) achieved through **parameter conversion**. At its core, parameter conversion allows *translation* of data stored in one data structure (e.g. an ImgLib2 ``RandomAccessibleInterval``) into a different data structure (e.g. an OpenCV ``Mat``) **on the fly**. This allows SciJava Ops to execute Ops backed by OpenCV code **on ImgLib2 data structures**.
+
+.. figure:: https://media.scijava.org/scijava-ops/1.0.1/parameter-conversion-opencv.svg
+
+At matching time, parameter conversion is invoked when an Op matches a user request in name and in Op type, but differing in individual parameter types. In these situations, it looks for ``engine.convert`` Ops that could potentially convert the user's provided inputs into the required Op inputs, and the same, in the other direction, for the output.
+
+A toy example
+=============
+
+Suppose we have an Op that inherently operates on ``RandomAccessibleInterval<DoubleType>``\ s:
+
+.. code-block:: java
+
+	/**
+	 * Conolves an image with a kernel, returning the output in a new object
+	 *
+	 * @param input the input data
+	 * @param kernel the kernel
+	 * @return the convolution of {@code input} and {@code kernel}
+	 * @implNote op names="filter.convolve"
+	 */
+	public static RandomAccessibleInterval<DoubleType> convolveNaive(
+			final RandomAccessibleInterval<DoubleType> input,
+			final RandomAccessibleInterval<DoubleType> kernel
+        ) {
+            // convolve convolve convolve //
+        }
+
+This Op might work well, however if users have a small kernel that is *only* used for this Op, they may find it frustrating to represent that data as a ``RandomAccessibleInterval``. Fortunately, SciJava Ops allows us to write ``engine.convert`` Ops, so users can pass their data in a data structure more convenient for them.
+
+
+.. code-block:: java
+
+    Img<DoubleType> in = ...
+    // 3x3 averaging kernel
+    double m = 1/9;
+    double[] data = new double[] { //
+        m, m, m, //
+        m, m, m, //
+        m, m, m //
+    };
+    // Not ideal
+    Img<DoubleType> kernel = ArrayImgs.doubles(data, new long[] {3, 3});
+
+    var result = ops.op("filter.convolve").input(in, kernel).apply();
+
+In this case, users might find it nicer to specify their kernel as a ``double[][]``, which is much easier for users to construct
+
+.. code-block:: java
```

Comment:
I think I would put this block after the converter definition, to show what is now possible with the converter